Revise the DS section (#87)
mlswg/mls-architecture: draft-ietf-mls-architecture.md

* A credential from the Authentication Service attesting to the
  binding between the identity and the client's signature key.

* The client's asymmetric encryption public key material.

All the parameters in the KeyPackage are signed with the signature private key

> corresponding to to the credential.

Reviewer: ```suggestion
corresponding to the credential.
```

### Delivery of messages and attachments {#delivery-guarantees}

The main responsibility of the Delivery Service is to ensure delivery of
messages. Some MLS messages need only be delivered to some members of group
(e.g., the message initializing a new member's state), while others need to be
delivered to all members.  The Delivery Service may enable these delivery
patterns via unicast channels (sometimes known as "client fanout"), broadcast
channels ("server fanout"), or a mix of both.

For the most part, MLS does not require the Delivery Service to deliver messages
in any particular order.  The one requirement is that because an MLS group has a
linear history, the members of the group must agree on the order in which
changes are applied.  Concretely, the group must agree on which MLS Commit
messages to apply.  There are a variety of ways to achieve this agreement, but

> most of them rely on some help from the Delivery Service.  For example, if a

Reviewer: If we decide to cast the DS as an abstract layer rather than a service (as done below), we could mandate that the DS does actually have to deal with the problem of deciding which commit to merge.

Delivery Service provides delivery in the same order to all group members, then
the members can simply apply Commits in the order in which they appear.

For the most part, MLS does not require the Delivery Service to deliver messages
in any particular order.  The one requirement is that because an MLS group has a
linear history, the members of the group must agree on the order in which
changes are applied.  Concretely, the group must agree on which MLS Commit

> messages to apply.

Reviewer: I think the above paragraph is a dupe.

Reviewer 2: I agree.

Each Commit is premised on a given state or "epoch" of the group.  The Delivery
Service must transmit to the group exactly one Commit message per epoch.

Much like the Authentication Service, the Delivery Service can be split between

> server and client components. Acheiving required uniqueness property will

Reviewer: ```suggestion
server and client components. Achieving the required uniqueness property will
```

It might make sense to lead with the statement that the DS can be split between client and server. Right now, the paragraphs above here read like it's describing a service running on a server.

typically require a combination of client and server behaviors.  For example,

> both of the following examples provide a unique Commit per epoch:

Reviewer: ```suggestion
all of the following examples provide a unique Commit per epoch:
```

> * An "filtering server" Delivery Service where a server rejects all but the

Reviewer: ```suggestion
* A "filtering server" Delivery Service where a server rejects all but the
```

  first Commit for an epoch and clients apply each Commit they receive.

> * An "ordered server" Delivery Service where a server forwards all messages

Reviewer: ```suggestion
* An "ordering server" Delivery Service where a server forwards all messages
```

  but assures that all clients see Commits in the same order, and clients.

> * A "passive server" Delivery Service where a servier forwards all messages

Reviewer: ```suggestion
* A "passive server" Delivery Service where a server forwards all messages
```

membership list is stored on some server associated with the Delivery
Service.

While this knowledge is not a breach of the protocol's authentication or
confidentiality guarantees, it is a serious issue for privacy.  Applications

> should consider anonymous systems for server fanout such as Loopix {{Loopix}}.

Reviewer: I thought this paragraph was w.r.t. the membership list, so encryption at rest makes sense. Maybe include the link to Loopix in the sentence that mentions traffic analysis?

Reviewer 2: Will re-introduce
